source/common/upstream/cluster_manager_impl.cc

```diff
@@ -422,6 +425,40 @@ void ClusterManagerImpl::onClusterInit(Cluster& cluster) {
   // We have a situation that clusters will be immediately active, such as static and primary
   // cluster. So we must have this prevention logic here.
   if (cluster_data != warming_clusters_.end()) {
+    if (cluster.info()->transportSocket().has_typed_config()) {
```

Comment:
This shouldn't be part of cluster manager, it breaks the transport socket abstraction a lot by assuming UpstreamTlsContext is the only TLS impl and how TLS transport socket will behave.

Instead, we should use the ready callback added in #13516, or add some `isready` method in transport socket factory, and check whether secret is ready or not.

Reply:
Done. like this?